wrapper.js

```diff
@@ -40,54 +51,65 @@ function setupMethods (soljson) {
     var removeFunction = soljson.removeFunction || soljson.Runtime.removeFunction;
 
     var cb = addFunction(wrapCallback(readCallback));
+    args.push(cb);
+    var cbSmt;
+    if (smtCallback !== undefined) {
+      cbSmt = addFunction(wrapCallback(smtCallback));
+      args.push(cbSmt);
+    }
     var output;
     try {
-      args.push(cb);
       output = compile.apply(undefined, args);
     } catch (e) {
       removeFunction(cb);
+      if (cbSmt !== undefined) {
+        removeFunction(cbSmt);
+      }
       throw e;
     }
     removeFunction(cb);
+    if (cbSmt !== undefined) {
+      removeFunction(cbSmt);
+    }
     return output;
   };
 
   var compileJSON = null;
   if ('_compileJSON' in soljson) {
-    compileJSON = soljson.cwrap('compileJSON', 'string', ['string', 'number']);
+    compileJSON = soljson.cwrap('compileJSON', 'string', ['string', 'number', 'number']);
   }
 
   var compileJSONMulti = null;
   if ('_compileJSONMulti' in soljson) {
-    compileJSONMulti = soljson.cwrap('compileJSONMulti', 'string', ['string', 'number']);
+    compileJSONMulti = soljson.cwrap('compileJSONMulti', 'string', ['string', 'number', 'number']);
   }
 
   var compileJSONCallback = null;
   if ('_compileJSONCallback' in soljson) {
-    var compileInternal = soljson.cwrap('compileJSONCallback', 'string', ['string', 'number', 'number']);
+    var compileInternal = soljson.cwrap('compileJSONCallback', 'string', ['string', 'number', 'number', 'number']);
     compileJSONCallback = function (input, optimize, readCallback) {
       return runWithReadCallback(readCallback, compileInternal, [ input, optimize ]);
     };
   }
 
   var compileStandard = null;
   if ('_compileStandard' in soljson) {
-    var compileStandardInternal = soljson.cwrap('compileStandard', 'string', ['string', 'number']);
+    var compileStandardInternal = soljson.cwrap('compileStandard', 'string', ['string', 'number', 'number']);
     compileStandard = function (input, readCallback) {
       return runWithReadCallback(readCallback, compileStandardInternal, [ input ]);
     };
   }
   if ('_solidity_compile' in soljson) {
-    var solidityCompile = soljson.cwrap('solidity_compile', 'string', ['string', 'number']);
+    var solidityCompile = soljson.cwrap('solidity_compile', 'string', ['string', 'number', 'number']);
```

Comment:
We cannot just change this. We need to define a new function name. This breaks backwards compatibility. There doesn't seem to be a way to detect the signature of a given symbol. Well we could detect the signature based on what `solidity_version` returns.